Marlin/src/core/types.h

```diff
@@ -72,6 +72,12 @@ struct IF<true, L, R> { typedef L type; };
 //
 typedef float feedRate_t;
 
+//
+// celsius_t is the native unit of temperature. Signed to handle a disconnected thermistor value (-14).
+// For more resolition (e.g., for a chocolate printer) this may later be changed to Celsius x 100
+//
+typedef int16_t celsius_t;
```

Comment:
If you want the compiler to spot any usage error (likely silent conversion from whatever type to celsius_t), you should declare a wrapper here:
```
class Celsius {
   int16_t temp;
public:
   inline operator int16_t () const { return temp; }
   inline Celsius & operator =(const Celsius & o) { temp = o.temp; return *this; }
   
   explicit Celsius(const int16_t value) : temp(value) {} // Make it explicit so it's clear that a value is converted to a Celsius in the code
   Celsius(const fahrenheit_t v): temp((v - 32) * 0.5555555556f) {} // Safe conversion from Fahrenheit to Celsius
   Celsius(const kelvin_t v) : temp(v - 273.15f) {} // Safe conversion from Kelvin to Celsius
private:
   Celsius(const double); // Prevent creating a Celsius from a floating point value, since the unit is unknown
};
typedef Celsius celcius_t;
```

Reply:
"It is the future you see." — Yoda

To get syntax coloring, the trick is to start with ` ```cpp `, ` ```diff `, etc….